src/pybind/mgr/dashboard/controllers/grafana.py

```diff
@@ -46,11 +49,19 @@ def instance(cls):
         :rtype: GrafanaRestClient
         """
         if not cls._instance:
-            url = Settings.GRAFANA_API_URL
-            user = Settings.GRAFANA_API_USERNAME
-            password = Settings.GRAFANA_API_PASSWORD
-
-            cls._instance = GrafanaRestClient(url, user, password)
+            kwargs = {}
+            if Settings.GRAFANA_API_AUTH_METHOD.lower() == 'password':
+                kwargs['username'] = Settings.GRAFANA_API_USERNAME
+                kwargs['password'] = Settings.GRAFANA_API_PASSWORD
+            elif Settings.GRAFANA_API_AUTH_METHOD.lower() == 'token':
+                kwargs['token'] = Settings.GRAFANA_API_TOKEN
+            else:
+                raise LookupError('No or unknown authentication method '
+                                  'provided. Please consult the documentation '
+                                  'about how to configure Grafana for the '
```

Comment:
Is it Grafana or the dashboard itself that need to be configured correctly in this case?

Reply:
It's the dashboard which needs to be configured correctly to be able to access Grafana. This error is raised if the authentication method in the dashboard is literally neither `password` nor `username`.